### ISSUE TITLE
small number formats don't seem to be respected

### ISSUE THREAD

**Issue by runfaj** [amcharts/amcharts4#1721]
Similar to: https://github.com/amcharts/amcharts4/issues/1142, but looks like maybe a new bug.

Copying the same code example from the above issue, then adding the `#.##a` number format doesn't seem to respect lower numbers. Instead of getting something like `.8, .6, .4` it displays as `800m, 600m, 400m`. Also seeing this in many other charts we're working with where increments do something like `0, 500m, 1, 1.5, 2, 2.5, ...`.

https://stackblitz.com/edit/amcharts-number-formatter-ywekmm?file=src/app/app.component.ts

**Comment by martynasma** [comment]
Well, "a" affects small numbers and big numbers.

By default, a small number is any number that's less than 1.

This is why when you enable "a" modifier 0.5 is formatted as 500m while 1, 1.5 remain as they are.

I suppose we could make that threshold (which is now hardcoded to 1) configurable.

E.g. consider number small if its less than X.

Do you think that could work for you?

**Comment by runfaj** [comment]
@martynasma Configuring the threshold would be super useful.

There's also times where we would want to convert something very small (like 5+ decimal places) and not convert anything between that and 1, but then continue to convert above 1. Would it be possible to not only allow changing that threshold, but also allowing the bigNumberFormat to accept decimal numbers? For example, maybe I would have
```
{ "number": 1e-3, "suffix": "m" },
{ "number": 1e-1, "suffix": "" },
{ "number": 0, "suffix": "" },
{ "number": 1, "suffix": "" },
{ "number": 1e3, "suffix": "k" },
```

**Comment by martynasma** [comment]
Hm, once you have a configurable small number threshold, what you had above should work automatically.

Making a note to try that after threshold is implemented.

**Comment by martynasma** [comment]
Fixed in [4.6.8](https://github.com/amcharts/amcharts4/releases/tag/4.6.8).

## [4.6.8] - 2019-10-07

### Added
- New `Export` property: [`validateSprites`](https://www.amcharts.com/docs/v4/reference/export/#validateSprites_property). It's an array you can push any `Sprite` to it. When export starts, it will check if those elements are not invalid and will await for them to be ready. This is useful if you need to do any modifications to your chart for export.
- New option for CSV and XSLX export options: `pivot`. If set to `true`, will "pivot" the data so that firsst column is field names with values lined up horizontally in rows.
- New property on `NumberFormatter`: `smallNumberThreshold` (default `1`). You can now set what is considered a "small number" when applying small number prefixes if `"a"` format modifier is used.

### Fixed
- If `ForceDirectedTree` was used with `am4core.options.onlyShowOnViewport = true`, the forces on links were not being calculated properly, resulting in incorrect layout.
- JSON: String dates will now be correctly parsed using `DateFormatter` in axis ranges' `date` and `endDate` properties.
- `WordCloud` was not properly parsing Korean text.
- If `MapChart` was zoomed in while still playing pan inertia animation, it produced a jerky effect.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by martynasma** [comment]
The "non-formatted" gap you mentioned can be achieved by setting `smallNumberThreshold` to a very low number as well as modifying `bigNumberPrefixes` to kick in at numbers you want them to kick in.
